api/android/api/src/main/jni/nnstreamer-native-singleshot.c

```diff
@@ -227,24 +228,35 @@ nns_native_single_invoke (JNIEnv * env, jobject thiz, jlong handle, jobject in)
 
   if (!nns_parse_tensors_data (pipe_info, env, in, FALSE, &in_data, NULL)) {
     nns_loge ("Failed to parse input tensors data.");
+    failed = TRUE;
     goto done;
   }
 
-  if (ml_single_invoke (single, in_data, &out_data) != ML_ERROR_NONE) {
-    nns_loge ("Failed to get the result from pipeline.");
+  /* create output object and get the direct buffer address */
+  if (!nns_create_tensors_data_object (pipe_info, env, priv->out_info_obj, &result) ||
+      !nns_parse_tensors_data (pipe_info, env, result, FALSE, &out_data, NULL)) {
+    nns_loge ("Failed to create output tensors object.");
+    failed = TRUE;
     goto done;
   }
 
-  if (!nns_convert_tensors_data (pipe_info, env, out_data, priv->out_info_obj,
-          &result)) {
-    nns_loge ("Failed to convert the result to data-object.");
-    result = NULL;
+  if (ml_single_invoke_no_alloc (single, in_data, out_data) != ML_ERROR_NONE) {
+    nns_loge ("Failed to invoke the model.");
+    failed = TRUE;
+    goto done;
   }
 
 done:
-  /* do not free input tensors (direct access from object) */
+  if (failed) {
+    if (result) {
```

Comment:
What about reducing just one line?
```cpp
if (failed && result) {
```

Reply:
@again4you Please remember and apply it after the release or at for-next :)